pkg/machines/components/vm/overview/bootOrder.jsx

```diff
@@ -360,26 +326,17 @@ function getBootOrder(vm) {
 
 export const BootOrderLink = ({ vm, idPrefix, close, dispatch, nodeDevices }) => {
     const [bootOrderShow, setBootOrderShow] = useState(false);
-    const bootOrderChanged = () => {
-        const activeDevices = getBootOrderDevices(vm);
-        const inactiveDevices = getBootOrderDevices(vm.inactiveXML);
-
-        // check if number bootable devices has changed
-        if (inactiveDevices.length !== activeDevices.length)
-            return true;
-        else
-            // check if boot order of any device has changed
-            return !inactiveDevices.every((element, index) => element.bootOrder === activeDevices[index].bootOrder);
-    };
+    const modalButton = (
+        <Button variant="link" className="edit-inline" isInline isAriaDisabled={vm.state != 'shut off'} onClick={setBootOrderShow}>
+            {_("edit")}
+        </Button>
+    );
 
     return (
         <>
             {bootOrderShow && <BootOrderModal close={() => setBootOrderShow(false)} vm={vm} dispatch={dispatch} nodeDevices={nodeDevices} />}
             {getBootOrder(vm)}
-            { vm.persistent && vm.state === "running" && bootOrderChanged() && <WarningInactive iconId="boot-order-tooltip" tooltipId="tip-boot-order" /> }
-            <Button variant="link" className="edit-inline" isInline isDisabled={!vm.persistent} onClick={setBootOrderShow}>
-                {_("edit")}
-            </Button>
+            {vm.state == 'shut off' ? modalButton : <Tooltip content={_("Only editable when the guest is shut off")}>{modalButton}</Tooltip>}
```

Comment:
Isn't the only problem just with showing the warning? So wouldn't it solve the problem if we just get rid of the warning which does not work, but still allow to edit running guests?

Reply:
It's only the warning problematic indeed. What you suggest would be ideal, I just thought it's too much work. RIght now we compare devices betwen the inactive and active XML, according to the index (order) in which they are presented. That's wrong, because devices can be added, removed, then the order changes. So a solution would be indeed to have a device comparison function between the two XMLs. Is it something that you 're interested to work on?

(the only reason I removed live editing is because I was lazy (busy with other things) to implement the functionality properly.

Reply:
> (the only reason I removed live editing is because I was lazy (busy with other things) to implement the functionality properly.

But this functionality is implemented properly, no?
In pkg/machines/libvirt-xml-update.js there is a function updateBootOrder . When it wants to change a boot order of some device, it tries to find the proper device by various identifiers (for disks it's a target, for a network it's a MAC, ....). It does not just find the device by its index. 
And also transient devices from active XML should not bother us for live editing, because only devices visible in Boot Order Modal are the permanent devices (I tried to add transient disk to a running VM, it was not visible in boot order modal). And when boot order is updated, only inactive XML is updated.

So If I'm not wrong and above mentioned is correct, I propose:
- Let's keep the live editing. 
- Let's keep the warning in the boot order modal. Since we are dealing only with permanent devices and inactive XML, any change would need a VM reboot.
- Let's remove warning icon in overview card. That one is truly broken in some cases.

Reply:
All what you describe above is correct. I just thought it's weird for every other attribute to show the warning triangle in the overview, but not for the bootorder.

Then transient disks and networks which show up in the respective lists, will also not appear in the boot order dialog, which might cause the users to get confused.

Reply:
Okay, makes sense to keep consistent. 
Also considering that live edit doens't matter that much here, since user has to turn of the machine anyway to use the functionality of the boot order, so I'm not so keen on having the live edit functionality.

Reply:
Right I was thinking the same thing.